docs/observability/index.mdx

```diff
@@ -0,0 +1,301 @@
+
+## 3. Set up your environment
+
+<CodeTabs
+  tabs={[
+    ShellBlock(`export LANGCHAIN_TRACING_V2=true
+export LANGCHAIN_API_KEY="<your-langchain-api-key>"
+# The example uses OpenAI, but it's not necessary if your code uses another LLM provider
+export OPENAI_API_KEY="<your-openai-api-key>"`),
+  ]}
+  groupId="client-language"
+/>
+
+## 4. Define your application
+
+We will use a simple [RAG](https://www.mckinsey.com/featured-insights/mckinsey-explainers/what-is-retrieval-augmented-generation-rag)
+application for this tutorial, but feel free to use your own code if you'd like - just make sure
+it has an LLM call!
+
+<details>
+  <summary>Application Code (Python and TypeScript)</summary>
+  <CodeTabs
+    groupId="client-language"
+    tabs={[
+      python({ label: "Python" })`
+      from openai import OpenAI
+      
+      openai_client = OpenAI()
+      
+      # This is the retriever we will use in RAG
+      # This is mocked out, but it could be anything we want
+      def retriever(query: str):
+          results = ["Harrison worked at Kensho"]
+          return results
+      
+      # This is the end-to-end RAG chain.
+      # It does a retrieval step then calls OpenAI
+      def rag(question):
+          docs = retriever(question)
+          system_message = """Answer the users question using only the provided information below:
+          
+          {docs}""".format(docs="\\n".join(docs))
+          
+          return openai_client.chat.completions.create(
+              messages=[
+                  {"role": "system", "content": system_message},
+                  {"role": "user", "content": question},
+              ],
+              model="gpt-4o-mini",
+          )
+    `,
+      typescript({ label: "TypeScript" })`
+      import { OpenAI } from "openai";
+      
+      const openAIClient = new OpenAI();
+      
+      // This is the retriever we will use in RAG
+      // This is mocked out, but it could be anything we want
+      async function retriever(query: string) {
+        return ["This is a document"];
+      }
+      
+      // This is the end-to-end RAG chain.
+      // It does a retrieval step then calls OpenAI
+      async function rag(question: string) {
+        const docs = await retriever(question);
+        
+        const systemMessage =
+          "Answer the users question using only the provided information below:\\n\\n" +
+          docs.join("\\n");
+          
+        return await openAIClient.chat.completions.create({
+          messages: [
+            { role: "system", content: systemMessage },
+            { role: "user", content: question },
+          ],
+          model: "gpt-4o-mini",
+        });
+      }
+    `,
+    ]}
+  />
+</details>
+
+## 5. Trace OpenAI calls
+
+The first thing you might want to trace is all your OpenAI calls. LangSmith makes this easy with the `wrap_openai/wrapOpenAI` wrappers.
```

Comment:
add links to SDK docs

https://docs.smith.langchain.com/reference/python/wrappers/langsmith.wrappers._openai.wrap_openai
https://docs.smith.langchain.com/reference/js/functions/wrappers_openai.wrapOpenAI